pyanaconda/modules/subscription/subscription.py

```diff
@@ -719,6 +761,77 @@ def parse_attached_subscriptions_with_task(self):
         )
         return task
 
+    def download_satellite_provisioning_script_with_task(self):
+        """Download Satellite provisioning script with task.
+
+        :return: a DBus path of an installation task
+        """
+        # construct proxy URL needed by the task from the
+        # proxy data in subscription request (if any)
+        # (it is logical to use the same proxy for provisioning
+        #  script download as for RHSM access)
+        if self.subscription_request.server_proxy_hostname:
```

Comment:
My idea was to limit the amount of things passed to the task, and there are up to 4 values needed to construct the proxy URL. But I guess I can just pass the subscription request instead, so that removes that concern and the code can be moved.

Reply:
Sounds like a job for a "structure" type of thing?

(I haven't reviewed the actual code yet, just sampling the comments so far. So I might be wrong.)